opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/authorization/ServiceSessionAndAuthSessionTest.java

```diff
@@ -0,0 +1,244 @@
+                .when(xs2aUpdateAuthorization)
+                .execute(any());
+
+        FacadeRedirectErrorResult errorResponse = (FacadeRedirectErrorResult) updateAuthorizationService.execute(buildAuthRequest(listAccountsResponse)).get();
+
+        assertThat(UUID.fromString(errorResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(errorResponse.getServiceSessionId())).isEqualTo(sessionId);
+        assertThat(errorResponse.getXRequestId()).isEqualTo(REQUEST_ID);
+        assertThat(errorResponse.getRedirectionTo().toString()).isEqualTo(REDIRECT_URL_NO_OK);
+        assertThat(errorResponse.getHeaders().get(ResponseHeaders.X_ERROR_CODE)).isEqualTo(errorResult.getCode());
+        assertThat(errorResponse.getHeaders().get(ResponseHeaders.X_ERROR_MESSAGE)).isEqualTo(errorResult.getMessage());
+
+        assertAllSessions(sessionId);
+    }
+
+    private AuthorizationRequest buildAuthRequest(FacadeStartAuthorizationResult listAccountsResponse) {
+        Map<String, String> authData = new HashMap<>();
+        authData.put("consent.frequencyPerDay", "5");
+        authData.put("psuIpAddress", "1.1.1.1"); // NOPMD Hard code the IP address in test is not a problem
+        authData.put("psuId", "anton.brueckner");
+        authData.put("consent.validUntil", "2020-02-26T22:00:00.000Z");
+        authData.put("consent.access.allAccountsAccess", "allAccountsWithBalances");
+
+        return AuthorizationRequest.builder()
+                       .facadeServiceable(FacadeServiceableRequest.builder()
+                                                  .redirectCode(listAccountsResponse.getRedirectCode())
+                                                  .authorizationSessionId(listAccountsResponse.getAuthorizationSessionId())
+                                                  .requestId(listAccountsResponse.getXRequestId())
+                                                  .build()
+                       )
+                       .scaAuthenticationData(authData)
+                       .build();
+    }
+
+    private ListAccountsRequest buildListAccountRequest(UUID sessionId) {
+        return ListAccountsRequest.builder()
+                       .facadeServiceable(
+                               FacadeServiceableRequest.builder()
+                                       .bankId(TEST_BANK_ID)
+                                       .requestId(REQUEST_ID)
+                                       .serviceSessionId(sessionId)
+                                       .sessionPassword(PASSWORD)
+                                       .fintechRedirectUrlOk(REDIRECT_URL_OK)
+                                       .fintechRedirectUrlNok(REDIRECT_URL_NO_OK)
+                                       .build()
+                       ).build();
+    }
+
+    private void assertAllSessions(UUID sessionId) {
+        Optional<AuthSession> authenticationSessionOptional = authenticationSessions.findById(sessionId);
```

Comment:
IMHO directly unwrapping Optional in test with `get()` is shorter and more concise